tools/sigclient/pkg/query/validation.go

```diff
@@ -0,0 +1,664 @@
+	if response == nil {
+		return
+	}
+	res.TotalMatched = response["totalMatched"]
+}
+
+func populateTotalMatchedAndRecords(response map[string]interface{}, res *Result) error {
+	populateTotalMatched(response, res)
+	return populateRecords(response, res)
+}
+
+func getHits(response map[string]interface{}) (map[string]interface{}, error) {
+	_, exist := response["hits"]
+	if exist && response["hits"] != nil {
+		hits, isMap := response["hits"].(map[string]interface{})
+		if !isMap {
+			return nil, fmt.Errorf("getHits: hits is not a map, received type: %T", response["hits"])
+		}
+		return hits, nil
+	}
+	return nil, nil
+}
+
+// Evaluates queries and compares the results with expected results for websocket
+func EvaluateQueryForWebSocket(dest string, queryReq map[string]interface{}, qid int, expRes *Result) error {
+	webSocketURL := fmt.Sprintf("ws://%s/api/search/ws", dest)
+	query := queryReq["searchText"].(string)
+
+	// create websocket connection
+	conn, _, err := websocket.DefaultDialer.Dial(webSocketURL, nil)
+	if err != nil {
+		return fmt.Errorf("EvaluateQueryForWebSocket: Error connecting to WebSocket server, webSocketURL: %v, err: %v", webSocketURL, err)
+	}
+	defer conn.Close()
+
+	err = conn.WriteJSON(queryReq)
+	if err != nil {
+		return fmt.Errorf("EvaluateQueryForWebSocket: Received error from server, queryReq: %v, err: %+v\n", queryReq, err)
+	}
+
+	readEvent := make(map[string]interface{})
+	sTime := time.Now()
+	tempRes := &Result{}
+	for {
+		err = conn.ReadJSON(&readEvent)
+		if err != nil {
+			log.Infof("EvaluateQueryForWebSocket: Received error from server, queryReq: %v, err: %+v\n", queryReq, err)
+			break
+		}
+		switch readEvent["state"] {
+		case "RUNNING":
+		case "QUERY_UPDATE":
+			// As query results come in chunks, we need to keep track of all the records
+			hits, err := getHits(readEvent)
+			if err != nil {
+				return fmt.Errorf("EvaluateQueryForWebSocket: Error getting hits, readEvent: %v, err: %v", readEvent, err)
+			}
+			err = populateRecords(hits, tempRes)
+			if err != nil {
+				return fmt.Errorf("EvaluateQueryForWebSocket: Error populating records, hits: %v, readEvent: %v, err: %v", hits, readEvent, err)
+			}
+		case "COMPLETE":
+			queryRes, err := CreateResult(readEvent)
+			if err != nil {
+				return fmt.Errorf("EvaluateQueryForWebSocket: Error creating result, readEvent: %v, err: %v", readEvent, err)
+			}
+			queryRes.Records = tempRes.Records
+			populateTotalMatched(readEvent, queryRes)
+
+			err = CompareResults(queryRes, expRes, query)
+			if err != nil {
+				return fmt.Errorf("EvaluateQueryForWebSocket: Failed evaluating query: %v, readEvent: %v, err: %v", query, readEvent, err)
+			}
+		default:
+			return fmt.Errorf("EvaluateQueryForWebSocket: Received unknown message from server, readEvent: %+v\n", readEvent)
+		}
+	}
+	log.Infof("EvaluateQueryForWebSocket: Query %v was succesful. In %+v", query, time.Since(sTime))
```

Comment:
same comment as above

Reply:
With %v parameter, it would show time with unit.
For example:

INFO[0030] EvaluateQueryForWebSocket: Query app_name=Bracecould | fields app_name, batch, http_status, ident was succesful. In 11.285875ms 
INFO[0030] EvaluateQueryForAPI: Query app_name=Bracecould | fields app_name, batch, http_status, ident was succesful. In 8.570875ms 
I don't think there is a point in using humanize.Comma since, the query execution be done in ms usually (10-200 ms) and if the value is larger, the %v should handle this by itself.